### PR TITLE
Declarative Shadow DOM should be exposed in ElementInternals.shadowRoot

### PR DESCRIPTION
**Pull request by rniwa**
#### 198be334f155e62044fdf1a2349d08db055a3423
<pre>
Declarative Shadow DOM should be exposed in ElementInternals.shadowRoot
<a href="https://bugs.webkit.org/show_bug.cgi?id=249523">https://bugs.webkit.org/show_bug.cgi?id=249523</a>

Reviewed by Chris Dumez.

Expose the declarative shadow DOM in ElementInternals.prototype.shadowRoot.

* LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root-expected.txt: Added.
* LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html: Added.
* LayoutTests/platform/win/TestExpectations:
* Source/WebCore/dom/ElementInternals.cpp:
(WebCore::ElementInternals::shadowRoot const):

Canonical link: <a href="https://commits.webkit.org/258119@main">https://commits.webkit.org/258119@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/a01157b58cb6b467c41deaefd2d1ca7bc0cbe8b0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100952 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10105 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34002 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110255 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/170519 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104939 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11036 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/976 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93360 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108094 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106735 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8347 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91603 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34952 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90255 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23002 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77932 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3783 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24520 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3807 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/923 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9918 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44023 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5579 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2919 "Built successfully and passed tests") | | | |

### CODE REVIEW

File: LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            super();
+            internals = this.attachInternals();
+        }
+    });
+    assert_true(!!internals);
+    assert_true(!!internals.shadowRoot);
+    assert_equals(internals.shadowRoot.mode, 'closed');
+    assert_equals(internals.shadowRoot.innerHTML.trim(), '<slot>hello, world.</slot>');
+}, 'Declarative Shadow DOM in "closed" mode is exposed via ElementInternals.prototype.shaowRoot');
```
Comment (cdumez) [inline LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html]: Typo: shaowRoot

File: LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            super();
+            internals = this.attachInternals();
+        }
+    });
+    assert_true(!!internals);
+    assert_true(!!internals.shadowRoot);
+    assert_equals(internals.shadowRoot.mode, 'open');
+    assert_equals(internals.shadowRoot.innerHTML.trim(), '<slot>hello, world.</slot>');
+}, 'Declarative Shadow DOM in "open" mode is exposed via ElementInternals.prototype.shaowRoot');
```
Comment (cdumez) [inline LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html]: ditto.

**Review by cdumez — APPROVED** [review]
LGTM

File: LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html (hunk 2 above)
Comment (rniwa) [reply LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html]: Fixed.

File: LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html (hunk 1 above)
Comment (rniwa) [reply LayoutTests/fast/shadow-dom/element-internals-exposes-declarative-shadow-root.html]: Fixed.

**Pull request merged**